themes/classic/templates/checkout/_partials/cart-voucher.tpl

```diff
@@ -32,40 +32,35 @@
               {foreach from=$cart.vouchers.added item=voucher}
                 <li class="cart-summary-line">
                   <span class="label">{$voucher.name}</span>
-                  <a href="{$voucher.delete_url}" data-link-action="remove-voucher"><i class="material-icons">&#xE872;</i></a>
                   <div class="float-xs-right">
-                    {$voucher.reduction_formatted}
+                    <span>{$voucher.reduction_formatted}</span>
+                    <a href="{$voucher.delete_url}" data-link-action="remove-voucher"><i class="material-icons">&#xE872;</i></a>
                   </div>
                 </li>
               {/foreach}
             </ul>
           {/block}
         {/if}
-
         <p>
           <a class="collapse-button promo-code-button" data-toggle="collapse" href="#promo-code" aria-expanded="false" aria-controls="promo-code">
             {l s='Have a promo code?' d='Shop.Theme.Checkout'}
           </a>
         </p>
-
         <div class="promo-code collapse{if $cart.discounts|count > 0} in{/if}" id="promo-code">
           {block name='cart_voucher_form'}
             <form action="{$urls.pages.cart}" data-link-action="add-voucher" method="post">
               <input type="hidden" name="token" value="{$static_token}">
               <input type="hidden" name="addDiscount" value="1">
-              <input class="promo-input" type="text" name="discount_name" placeholder="{l s='Promo code' d='Shop.Theme.Checkout'}">
-              <button type="submit" class="btn btn-primary"><span>{l s='Add' d='Shop.Theme.Actions'}</span></button>
+              <input class="promo-input" type="text" name="discount_name" placeholder="{l s='Promo code' d='Shop.Theme.Checkout'}"><!-- Tricks: this comment sticks both element together
+              --><button type="submit" class="btn btn-primary"><span>{l s='Add' d='Shop.Theme.Actions'}</span></button>
```

Comment:
I think this can be done using CSS only

Reply:
Yes I know but it's a trick to avoid using CSS.
Is it a problem ?

Reply:
Well the purpose of CSS is precisely to avoid doing this kind of things.

Reply:
Okay, I will change it